pkg/recipes/configloader/secrets.go

```diff
@@ -35,10 +35,14 @@ type secretsLoader struct {
 	ArmClientOptions *arm.ClientOptions
 }
 
-// LoadSecrets loads secrets from secret stores based on input map of provided secret store IDs and secret keys.
-// It returns a map of secret data, where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
-func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDResourceKeys map[string][]string) (secretData map[string]map[string]string, err error) {
-	for secretStoreID, secretKeys := range secretStoreIDResourceKeys {
+// LoadSecrets loads secrets from secret stores based on input map of provided secret store IDs and secret keys filter.
+// If the input secret keys filter for a secret store ID is nil or empty, it retrieves secret data for all keys for that secret store.
+// When the secret keys filter is populated for a secret store ID, it retrieves secret data for the specified keys for the associated secret store ID.
+// The function returns a map of secret data, where the keys are the secret store IDs and the values are maps of secret keys and their corresponding values.
+func (e *secretsLoader) LoadSecrets(ctx context.Context, secretStoreIDKeysFilter map[string][]string) (secretData map[string]map[string]string, err error) {
```

Comment:
Why not just name this `filter` and give an example in the doc above?
```suggestion
func (e *secretsLoader) LoadSecrets(ctx context.Context, filter map[string][]string) (secretData map[string]map[string]string, err error) {
```